streams/src/test/java/org/apache/kafka/streams/integration/StandbyTaskCreationIntegrationTest.java

```diff
@@ -148,6 +161,82 @@ public void shouldCreateStandByTasksForMaterializedAndOptimizedSourceTables() th
         );
     }
 
+    @Test
+    public void shouldRecycleStateFromStandbyTaskPromotedToActiveTask() throws Exception {
```

Comment:
Wasn't really sure where to put this test, but this class seemed close enough. Verified that this does indeed fail on trunk without this fix

Reply:
This seems fine to me.